### PR TITLE
fix: put envs from .env files on process.env

### PR DESCRIPTION
**Pull request by sheremet-va**
Fixes https://github.com/vitest-dev/vitest/discussions/768

### CODE REVIEW

File: packages/vitest/src/node/plugins/index.ts [hunk 1]
```diff
@@ -62,14 +62,19 @@ export async function VitestPlugin(options: UserConfig = {}, ctx = new Vitest())
         )
         options.api = resolveApiConfig(options)
 
-        process.env.BASE_URL ??= viteConfig.base
-        process.env.MODE ??= viteConfig.mode
+        // we replace every "import.meta.env" with "process.env"
+        // to allow reassigning, so wee need to put all envs on process.env
```
Comment (patak-dev) [inline packages/vitest/src/node/plugins/index.ts:66]:
```suggestion
        // to allow reassigning, so we need to put all envs on process.env
```

**Review by patak-dev — APPROVED** [review]

File: test/core/test/env.test.ts [hunk 2]
```diff
@@ -1,6 +1,10 @@
 import { expect, test } from 'vitest'
 import { getAuthToken } from '../src/env'
 
+test('reads envs from .env file', () => {
+  expect(import.meta.env.VITE_TETS_ENV).toBe('local')
```
Comment (antfu) [inline test/core/test/env.test.ts:5]:
```suggestion
  expect(import.meta.env.VITE_TEST_ENV).toBe('local')
```

File: test/core/.env.local [hunk 3]
```diff
@@ -0,0 +1 @@
+VITE_TETS_ENV=local
```
Comment (antfu) [inline test/core/.env.local:1]:
```suggestion
VITE_TEST_ENV=local
```

**Review by antfu — APPROVED** [review]

**Comment by meoyawn** [comment]
hi folks! Has this been released? I'm trying to make `vitest` to pick up `.env.test` file, but can't find any instructions

**Pull request merged**